components/WorkSectionAsGallery.js

```diff
@@ -0,0 +1,221 @@
+import React, { PureComponent } from 'react';
+
+import Link from 'next/link';
+
+import PropTypes from 'prop-types';
+
+import { ContentfulMedia, SimpleFragment } from 'models';
+
+import { Markdown, Image } from 'components/base';
+
+import get from 'utils/get';
+import flattenImageData from 'utils/flattenImageData';
+
+const pauseStartVideo = (videoId) => {
+  const video = document.getElementById(videoId);
+  const videoButtonText = document.getElementById(`${videoId}--video-button-text`).textContent;
+
+  if (videoButtonText == '(Pause)') {
+    video.pause();
+    document.getElementById(`${videoId}--video-button-text`).textContent = '(Play)';
+  } else {
+    video.play();
+    document.getElementById(`${videoId}--video-button-text`).textContent = '(Pause)';
+  }
+}
```

Comment:
https://www.loom.com/share/f2c883d272844b86ba09bdfb60d218bf

Reply:
Basically it comes down to maintainability. If we store the video's "play" and "pause" state, it will be more maintainable down the road in case we want to have some other UI side effects based on the video's state. Right now it's just a text button -- what if down the road we want to render a play/pause icon, or add some visual effect, like a filter on the video? by storing the video state, we can just derive our UI from that state.

Reply:
I found this [article](https://dzone.com/articles/dom-manipulation-in-react) that mentions that it's slow to traverse through the DOM to find the element, so perhaps efficiency is another reason for opting to use React's Virtual DOM vs the regular DOM.